sky/clouds/aws.py

```diff
@@ -773,6 +784,38 @@ def get_user_identities(cls) -> Optional[List[List[str]]]:
         #   automatic switching for AWS. Currently we only support one identity.
         return [user_ids]
 
+    @classmethod
+    @functools.lru_cache(maxsize=1)  # Cache since getting identity is slow.
+    def get_user_identities(cls) -> Optional[List[List[str]]]:
```

Comment:
Let's add a docstr for this function for the behavior of caching and returning identity. It might also be good to move the docstr from `_sts_get_caller_identity` to here.